test/cql-pytest/test_allow_filtering.py

```diff
@@ -291,3 +291,23 @@ def test_contains_frozen_collection_ck(cql, test_keyspace):
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS 0 ALLOW FILTERING")))
         assert 1 == len(list(cql.execute(
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS KEY 0 ALLOW FILTERING")))
+
+# table4 contains example table from issue #8991
+@pytest.fixture(scope="module")
+def table4(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (k int, c1 int, c2 int, PRIMARY KEY (k,c1,c2))")
+    cql.execute(f"CREATE INDEX ON {table} (c2)")
+    for c1 in [0, 1, 2]:
+        for c2 in [0, 1, 2]:
+            cql.execute(f"INSERT INTO {table} (k, c1, c2) VALUES (0, {c1}, {c2})")
```

Comment:
Same comment I had in the previous patch - maybe you don't need 3*3 rows and can reproduce this bug with fewer and save a few milliseconds.

Reply:
I reduced it to just two rows.